mango-core/src/main/scala/org/bdgenomics/mango/models/TrackedLayout.scala

```diff
@@ -139,15 +191,47 @@ class OrderedTrackedLayout[T: ClassTag](values: Traversable[(ReferenceRegion, T)
   class Track(val initial: (ReferenceRegion, T)) {
 
     val records = new mutable.ListBuffer[(ReferenceRegion, T)]()
-    records += initial
+    val groupPairs = new mutable.ListBuffer[GroupPair]()
+    val idx: Int = numTracks
+
+    if (initial != null)
+      records += initial
+
+    def this() {
+      this(null)
+    }
 
     def +=(rec: (ReferenceRegion, T)): Track = {
       records += rec
       this
     }
 
+    def addGroupPair(recs: Seq[(ReferenceRegion, T)]) {
+      if (recs.size < 2)
```

Comment:
Does this mean that we drop unpaired reads? We shouldn't drop unpaired reads; instead we should just display the single read we get. Even with a paired end sequencing protocol, we can encounter situations where we just get a single read (you can have "chimeric" read pairs where one read maps to one chromosome, and the other read maps to a different chromosome).

Would this make the display/processing code significantly more complex? If so, perhaps let's drop these reads for now, but open an issue and come back and fix this later.